client-app/app-runner.ts

```diff
@@ -26,19 +26,27 @@ export default async (getPlugins: (options: { router: Router }) => { plugin: Plu
 
   const { fetchUser } = useUser();
   const { themeContext, fetchThemeContext } = useThemeContext();
-  const { currentLocale, currentLanguage, supportedLocales, setLocale } = useLanguages();
+  const { currentLocale, currentLanguage, supportedLocales, setLocale, fetchLocaleMessages } = useLanguages();
   const { currentCurrency } = useCurrency();
 
+  const fallback = {
+    locale: "en",
+    message: {},
+    async setMessage() {
+      this.message = await fetchLocaleMessages(this.locale);
+    },
+  };
+
   /**
    * Fetching required app data
    */
-  await Promise.all([fetchThemeContext(), fetchUser()]);
+  await Promise.all([fetchThemeContext(), fetchUser(), fallback.setMessage()]);
```

Comment:
I propose to expose `fetchFallbackLocale and fallbackLocale` from `useLanguages`, call it here, and pass to i18n.